db/database.go

```diff
@@ -1761,13 +1761,14 @@ func (db *DatabaseCollectionWithUser) resyncDocument(ctx context.Context, docid,
 			raw []byte, rawXattr []byte, deleteDoc bool, expiry *uint32, err error) {
 			// There's no scenario where a doc should from non-deleted to deleted during UpdateAllDocChannels processing,
 			// so deleteDoc is always returned as false.
-			if currentValue == nil || len(currentValue) == 0 {
-				return nil, nil, deleteDoc, nil, base.ErrUpdateCancel
-			}
 			doc, err := unmarshalDocumentWithXattr(docid, currentValue, currentXattr, currentUserXattr, cas, DocUnmarshalAll)
 			if err != nil {
 				return nil, nil, deleteDoc, nil, err
 			}
+			updatedHighSeq = doc.Sequence
+			if currentValue == nil || len(currentValue) == 0 {
+				return nil, nil, deleteDoc, nil, base.ErrUpdateCancel
+			}
```

Comment:
This doesn't quite seem like the right place for a fix, but I might be wrong. Why would we need to run `unmarshalDocumentWithXattr` before deciding that based on `currentValue`, we don't need to resync?

Notably: from `resyncDocument` we're now returning a "new sequence" that is actually the previous sequence!

I was thinking the loop fix would be inside _the caller_ of `resyncDocument` to appropriately handle zero-value sequences for the error conditions.

Reply:
https://github.com/couchbase/sync_gateway/pull/6346/commits/825508a9d99560ff2f431627d664ab234b473532 
Implemented the fix discussed in sync up yesterday, doc.sequence is now unmarhsalled from the QueryResync rows, with a check for views since the rows are different